### ISSUE TITLE
Script introduces errors into GCode - unusable output

### ISSUE THREAD

**Issue by eparadis** [Kranex/goskew#4]
Hi, first thanks for writing this. I love the idea and it is easy to use.

Unfortunately this script is introducing errors into the Gcode it processes.

Here is a screen shot of the last layer of a print as shown in Octoprint's Gcode viewer.
<img width="236" alt="gcode_no_errors" src="https://user-images.githubusercontent.com/2131850/187320849-14f347b7-adc5-4c01-8795-85a91583e58b.png">

Here is the same gcode after processing through `goskew`.
<img width="219" alt="gcode_errors" src="https://user-images.githubusercontent.com/2131850/187320820-e661556b-af4f-4b50-9906-ee8f35eb52ca.png">

You can clearly see the "hitches" in the second screenshot that are not supposed to be there. These are real - they do show up in the final print quite noticeably.

I'm not great at reading Gcode but I think you can see one of the hitches here:
```
;TYPE:External perimeter
G1 F1500
G1 X85.378 Y93.315 E.02112
G1 X85.074 Y92.903 E.02113
G1 X84.814 Y92.463 E.02112
G1 X84.594 Y92 E.02113
G1 X85 Y91.517 E.02112    // <--- note this line!
G1 X84.306 Y91.019 E.02112
G1 X84.235 Y90.512 E.02114
```

That should be the top left part of a circle, so you would expect the X coordinate to decrease smoothly, but that one lines has a "X85" for some unknown reason.

I think this is the same section in an unprocessed file:

```
;TYPE:External perimeter
G1 F1500
G1 X85.961 Y93.315 E.02112
G1 X85.655 Y92.903 E.02113
G1 X85.392 Y92.463 E.02112
G1 X85.172 Y92 E.02113
G1 X85 Y91.517 E.02112     // <--- original this line!
G1 X84.875 Y91.019 E.02112
G1 X84.8 Y90.512 E.02114
```

Perhaps there is an issue with integers being used instead of floating point? I don't know how `go` works with numerics.

I suspect this is an easy fix - perhaps forcing a type on whatever reads the input.

**Comment by eparadis** [comment]
I think the error is in the regex.

**Comment by eparadis** [comment]
I think the error is in the regex.

[This](https://github.com/Kranex/goskew/blob/47285f8ecbd1643ba7d5c4d3cadf2b42848d4cdc/goskew.go#L57) won't match an integer because it assumes a decimal point.
`\[xX](-?\d*\.\d*)`




Quick example cases [here](https://regex101.com/r/qL0U7D/1)

**Comment by eparadis** [comment]
this PR fixes my situation https://github.com/Kranex/goskew/pull/5

**Comment by Kranex** [comment]
@eparadis sorry I wasn't notified of this issue or the PR for some reason. I just had the most horrific print as a result of this, I'm amazed that I didn't notice this sooner as I skew literally everything. I'll merge in your PR now.

**Comment by Kranex** [comment]
merged and released to v1.3.1. Thank you for the fix!
